javascript/README.md

```diff
@@ -505,3 +505,32 @@ functionThatReturnsAPromise()
         // other code here!
     });
 ```
+
+## Module Definition
+
+When building a module that can be consumed by a number of different sources, we prefer
```

Comment:
What about modules that aren't consumed by a number of different sources... do we support a different module definition then?